src/shell.js

```diff
@@ -137,6 +137,11 @@ if (ENVIRONMENT_IS_NODE) {
     }
   });
 #endif
+  // Currently node will swallow unhandled rejections, but this behavior is
+  // deprecated, and in the future it will exit with error status.
+  process['on']('unhandledRejection', function(reason, p) {
+    process['exit'](1);
```

Comment:
how about adding before the exit
```
#if ASSERTIONS
    Module['printErr']('node.js exiting due to unhandledRejection'); // or some better text
#endif
```
? Maybe also printing the reason? Seems like it could help debugging.

Reply:
From my local testing, I saw the stacktrace from the rejection printed twice:
```
failed to asynchronously prepare wasm: CompileError: WasmCompile: Wasm decoding failed: expected magic word 00 61 73 6d, found 43 68 61 6e @+0
CompileError: WasmCompile: Wasm decoding failed: expected magic word 00 61 73 6d, found 43 68 61 6e @+0
    at /usr/local/google/home/dschuff/s/wasm-waterfall/hello.js:1634:28
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:678:11)
    at startup (bootstrap_node.js:187:16)
    at bootstrap_node.js:608:3
```
So that seemed sufficient. But maybe 'exiting due to unhandled rejection' would be useful.